packages/flutter/lib/src/isolate/isolate_worker.dart

```diff
@@ -0,0 +1,172 @@
+import 'dart:async';
+import 'dart:isolate';
+
+import '../../sentry_flutter.dart';
+import 'isolate_logger.dart';
+
+const _shutdownCommand = '_shutdown_';
+
+// -------------------------------------------
+// HOST-SIDE API (runs on the main isolate)
+// -------------------------------------------
+
+/// Minimal config passed to isolates - extend as needed.
+class WorkerConfig {
+  final bool debug;
+  final SentryLevel diagnosticLevel;
+  final String debugName;
+  final bool automatedTestMode;
+
+  const WorkerConfig({
+    required this.debug,
+    required this.diagnosticLevel,
+    required this.debugName,
+    this.automatedTestMode = false,
+  });
+}
+
+/// Host-side helper for workers to perform minimal request/response.
+/// Adapted from https://dart.dev/language/isolates#robust-ports-example
+class Worker {
```

Comment:
We could also re-use this for `_AndroidNativeReplayWorker`.

Reply:
yup, would be my next step after this

> Adjust AndroidReplayRecorder to use the new setup